resources-generator/src/test/kotlin/dev/icerock/gradle/generator/jsGenerator/JsMessageFormatTest.kt

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2024 IceRock MAG Inc. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package dev.icerock.gradle.generator.jsGenerator
+
+import dev.icerock.gradle.generator.platform.js.convertToMessageFormat
+import kotlin.test.Test
+import kotlin.test.assertEquals
+
+class JsMessageFormatTest {
+    @Test
+    fun cleanStringTest() {
+        assertEquals(
+            expected = "Clean text with 70% percents.",
+            actual = "Clean text with 70% percents.".convertToMessageFormat()
+        )
+    }
+
+    @Test
+    fun cleanStringWithDotInEndTest() {
+        assertEquals(
+            expected = "Only peoples who like scorpion choose 39%.",
+            actual = "Only peoples who like scorpion choose 39%.".convertToMessageFormat()
+        )
+    }
+
+    @Test
+    fun textWithSingleStringArgumentTest() {
+        assertEquals(
+            expected = "Same guys who like {0} choose pretty woman",
+            actual = "Same guys who like %s choose pretty woman".convertToMessageFormat()
+        )
+    }
+
+    @Test
+    fun textWithSingleDigitsArgumentTest() {
+        assertEquals(
+            expected = "Same guys who like {0} choose pretty woman",
+            actual = "Same guys who like %d choose pretty woman".convertToMessageFormat()
+        )
+    }
+
+    @Test
+    fun textWithSingleWithDecimalArgumentTest() {
+        assertEquals(
+            expected = "Only {0} percents of people like animals",
+            actual = "Only %.2d percents of people like animals".convertToMessageFormat()
+        )
+    }
```

Comment:
please add test with multiple not positional arguments

Reply:
added